cmake/simdjson-props.cmake

```diff
@@ -0,0 +1,47 @@
+#
```

Comment:
Mind a brief comment on the intent of this? I'm having trouble parsing the intent from the code.

Reply:
Since flags were previously accumulated via targets, those were causing issues with implementation details leaking into the install interface.

Now these commands will accumulate command calls into a string variable, then eventually flush them into a CMake script that is called with `include`.

Basically creates a file like this:
```cmake
target_compile_definitions("${target}" ${public} [==[arg1]==] [==[arg2]==])
target_compile_options("${target}" ${private} [==[arg1]==])
```

Reply:
Oh! So the idea is not to put these flags onto people who include simdjson? Hmm. It's a really good idea to separate them. I know a number of the flags are added on purpose, because they are needed when the user includes simdjson.h. Which ones should get removed?

Would a target we only use in simdjson (simdjson-internal or something) be a more idiomatic choice?

Reply:
@friendlyanon I think this is what `simdjson-internal-flags` was for.

The new approach does seem like a lot of extra code / concepts. But maybe the way we were doing it was abusing libraries, so that might be fine :)

Reply:
@friendlyanon checking: did I understand the purpose of this correctly? Did `simdjson-internal-flags` not cover this use case before?

As a naive cmake user it seems more complicated than `simdjson-internal-flags` is all.

Reply:
The problem with using targets for flags is that they will leak into the install interface.

Since there are tests, that essentially double the work by building the library again, I don't know if this could be removed. Maybe those tests could use some rework.